ztoc/fbs/ztoc09.fbs

```diff
@@ -0,0 +1,50 @@
+namespace v09;
+
+table Xattr {
+	key : string;
+	value : string;
+}
+
+table FileMetadata {
+	name : string;
+	type : string;
+	uncompressed_offset : long;
+	uncompressed_size : long;
+	linkname : string;		// Target name of link (valid for TypeLink or TypeSymlink)
+	mode : long;			// Permission and mode bits
+	uid : uint32;			// User ID of owner
+	gid : uint32;			// Group ID of owner
+	uname : string;			// User name of owner
+	gname : string;			// Group name of owner
+
+	mod_time  : string;		// Modification time (is formatted in RFC 3339 format,
+							// with sub-second precision added if present: "2006-01-02T15:04:05.999999999Z07:00")
+	devmajor : long;		// Major device number (valid for TypeChar or TypeBlock)
+	devminor : long;		// Minor device number (valid for TypeChar or TypeBlock)
+
+	xattrs : [Xattr];
+}
+
+enum CompressionAlgorithm : byte { Gzip = 1 }
+
+table CompressionInfo {
+	compression_algorithm : CompressionAlgorithm = Gzip;
+	max_span_id : int;			// The total number of spans in Ztoc - 1
+	span_digests : [string];
+	checkpoints : [ubyte];	// the binary data used to decompress the span
+}
+
+table TOC {
+	metadata : [FileMetadata];
+}
+
+table Ztoc {
+	version : string;			// The version of the Ztoc in format <major_version>.<minor_version>, e.g. 1.0
+	build_tool_identifier : string;
+	compressed_archive_size : long;
+	uncompressed_archive_size : long;
+	toc : TOC;
+	compression_info : CompressionInfo;
+}
```

Comment:
Is it possible to just add a new `table Ztoc09` (within the same fbs file) that without the `builder_tool_identifier` field? This can reduce a lot of duplicated code. And we're not changing anything else (e.g. `table CompressionInfo`/`TOC`/etc)

So the workflow will be:

for new data:

`Ztoc struct` (defined by SOCI)  <-> `Ztoc struct` (generated by flatbuf from `Ztoc flatbuf table`) <-> `Ztoc flatbuf bytes`

for old data (only unmarshal because we don't want to produce new data in old definition, i.e., `Ztoc09`):

`Ztoc09 flatbuf bytes` -> `Ztoc09 struct` (generated by flatbuf from `Ztoc09 flatbuf table`) -> `Ztoc struct` (defined by SOCI / just need to remove the builder tool field in `Ztoc09 struct`).

This also make it clean (only a new fbs file, only a new `Ztoc09` due to removal of `build_tool_identifier`) and clear(no changes on other tables like `CompressionInfo`/etc)

Reply:
I don't think we want these in the same flatbuffer definition because the goal is to keep v0.9 inside soci but move v1.0 to another repository eventually.

Reply:
Ztoc09 could be defined in a new file that includes this file.